### PR TITLE
mcs: solve forward stream error

### LINKED ISSUE #7320: The error of forwarding TSO should be handled
## Bug Report

https://github.com/tikv/pd/blob/ab8bf7b7a62d43abd9a8d5213fc3b5855472bd9e/server/grpc_service.go#L2626-L2653
https://github.com/tikv/pd/blob/ab8bf7b7a62d43abd9a8d5213fc3b5855472bd9e/server/grpc_service.go#L2635

The error of sending the request is not properly handled. We should do the error handling in case there is something wrong when forwarding the request to get TSO.

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #7320

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/grpc_service.go [hunk 1]
```diff
@@ -2632,7 +2632,13 @@ func (s *GrpcServer) getGlobalTSOFromTSOServer(ctx context.Context) (pdpb.Timest
 		if err != nil {
 			return pdpb.Timestamp{}, err
 		}
-		forwardStream.Send(request)
+		err := forwardStream.Send(request)
+		if err != nil {
+			s.tsoClientPool.Lock()
+			delete(s.tsoClientPool.clients, forwardedHost)
```
Comment (zeminzhou) [inline server/grpc_service.go:2638]: Do we need to do close？

**Review by rleungx — APPROVED** [review]

File: server/grpc_service.go (hunk 1 above)
Comment (lhy1024) [reply server/grpc_service.go:2638]: I  think so. We need to avoid that there are always errors.

**Review by JmPotato — APPROVED** [review]

**Pull request merged**